src/greeter/KeyboardModel.cpp

```diff
@@ -32,9 +34,14 @@ namespace SDDM {
 
     KeyboardModel::KeyboardModel() : d(new KeyboardModelPrivate) {
         if (QGuiApplication::platformName() == QLatin1String("xcb")) {
+#ifdef BUILD_WITH_X11
             m_backend = new XcbKeyboardBackend(d);
             m_backend->init();
             m_backend->connectEventsDispatcher(this);
+#else
+            qCritical("Xcb backend is not supported in this build");
+            std::terminate();
```

Comment:
Instead of terminating, it should warn and continue anyway.

Reply:
Fixed.